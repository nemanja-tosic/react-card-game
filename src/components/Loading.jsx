// @flow
import React from 'react';

import './Loading.css';

const Loading = ({message}: {message: string}) => (
  <div className="Loading">
    <i className="fa fa-cog fa-4x fa-spin"/>
    {message}
  </div>
);

export default Loading;
