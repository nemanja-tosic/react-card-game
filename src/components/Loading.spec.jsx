// @flow
import React from 'react';
import { shallow } from 'enzyme';

import Loading from './Loading';

describe('Loading', () => {
  let component;

  beforeEach(() => {
    component = shallow(<Loading message="some message" />);
  });

  it('should show the loading animation and the message', () => {
    expect(component.getNode()).toMatchSnapshot();
  });
});
