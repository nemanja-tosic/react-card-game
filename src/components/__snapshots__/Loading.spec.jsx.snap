// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should show the loading animation and the message 1`] = `
<div
  className="Loading"
>
  <i
    className="fa fa-cog fa-4x fa-spin"
  />
  some message
</div>
`;
